python/cuml/cuml/internals/input_utils.py

```diff
@@ -500,7 +499,7 @@ def convert_dtype(X, to_dtype=np.float32, legacy=True, safe_dtype=True):
     if the conversion would lose information.
     """
 
-    if isinstance(X, (dask_cudf.Series, dask_cudf.DataFrame)):
+    if hasattr(X, "__dask_graph__") and hasattr(X, "compute"):
```

Comment:
I think this might make this conditional more general than we want it. That said, this particular conditional was exclusively for dask-sql which is not an active project currently, so I would say we should keep the conditional to avoid any side effects and then look into removing it in 25.08 potentially

Reply:
I'm not sure I follow - the old conditional checked for instances of dask_cudf, this checks for any dask object. If you look at the PR where this was added, it seems pretty clear that the difference between this location and the other one is accidental. Any dask object should work fine here, and previously there wasn't a different code path that would be taken for other dask objects (things would just error).

Reply:
Going even further - a quick grep shows this function is only ever called once, as part of `cuml.benchmark`. Whole thing could probably be moved into the benchmarks or maybe just deleted wholesale.